core/src/avm2/object/xml_list_object.rs

```diff
@@ -438,4 +438,42 @@ impl<'gc> TObject<'gc> for XmlListObject<'gc> {
                 .unwrap_or(Value::Undefined))
         }
     }
+
+    fn delete_property_local(
+        self,
+        activation: &mut Activation<'_, 'gc>,
+        name: &Multiname<'gc>,
+    ) -> Result<bool, Error<'gc>> {
+        let mut write = self.0.write(activation.context.gc_context);
+
+        if !name.is_any_name() && !name.is_attribute() {
+            if let Some(local_name) = name.local_name() {
+                if let Ok(index) = local_name.parse::<usize>() {
+                    if index < write.children.len() {
+                        let removed = write.children.remove(index);
+                        let removed_node = removed.node();
+                        if let Some(parent) = removed_node.parent() {
+                            if let E4XNodeKind::Attribute(_) = &*removed_node.kind() {
+                                parent
+                                    .remove_attribute(activation.context.gc_context, &removed_node);
+                            } else {
+                                parent.remove_child(activation.context.gc_context, &removed_node);
+                            }
+                        }
+                    }
+                    return Ok(true);
+                }
+            }
+        }
+
+        for child in write.children.iter_mut() {
```

Comment:
What happens if I do any any attribute name (e.g. `@*`)?

Reply:
Works as you'd expect. I added that to the test to be safe.